src/net/udp/mod.rs

```diff
@@ -204,8 +224,12 @@ impl UdpSocket {
     /// ```
     pub async fn connect<A: ToSocketAddrs>(&self, addrs: A) -> io::Result<()> {
         let mut last_err = None;
+        let addrs = addrs
+            .to_socket_addrs()
+            .await
+            .context(|| String::from("could not resolve addresses"))?;
```

Comment:
It'd be great to know _which_ addresses couldn't be resolved. I feel this is one of the most common cases folks are left wondering about.